src/sandbox.rkt

```diff
@@ -268,21 +250,25 @@
       (rollback-improve!)
       (set! timeline (*timeline*))
       (when seed (set-seed! seed))
-      (with-handlers ([exn? (curry on-exception start-time)])
-        (define out
+      (define result
+        (with-handlers ([exn? (curry on-exception start-time)])
           (match command
             ['improve (run-herbie test)]
-            ['sample (get-sample test)]))
-        (print-warnings)
-        (if (eq? command 'sample) out (add-time out (- (current-inexact-milliseconds) start-time))))))
+            ['sample (get-sample test)])))
+      (print-warnings)
+      (define total-time (- (current-inexact-milliseconds) start-time))
+      (if (test-result? result)
+          (struct-copy test-result result [time total-time])
+          result)))
```

Comment:
Ok, if this is a different PR I'll accept that, but I think these 3 lines demonstrate an issue with the `test-result` design. Basically, the fields in `test-result` come from two thing:

1. Generic Herbie infrastructure such as the test, status, exception, timeline, warnings, and runtime.
2. Results specific to `run-herbie` such as the pcontexts, alts, etc.

I would rather split them up, so that we had a `job-result` struct with all the fields in (1) and then a `herbie-result` struct or whatever with the fields in (2). And then other jobs could reuse `job-result`, and these three lines could apply always because the result would always be a `job-result`.

Reply:
Oh, and the `job-result` fields could all be filled in in `get-test-result` instead of having to be filled in by `run-herbie`.

Reply:
Does this use inheritance? As in does `herbie-result` extend `job-result`?

Reply:
I experimented a bit. It might make sense for `job-result` to have a field called `backend` which contains any information from a backend, i.e., a `herbie-result` struct.

Reply:
Yeah that's what I meant. A `job-result` has a field called, like, `result` or something with the `herbie-result` or potentially a `pcontext` or an error score or a local error tree or whatever.